src/dropdown.ts

```diff
@@ -270,19 +266,32 @@ export default class Dropdown extends LitElement implements FormControl {
       }
     }
 
-    if (wasMultiple && this.lastSelectedAndEnabledOption?.value) {
+    if (
+      wasMultiple &&
+      this.lastSelectedAndEnabledOption?.label &&
+      this.lastSelectedAndEnabledOption?.value &&
+      this.#inputElementRef.value
+    ) {
       this.value = [this.lastSelectedAndEnabledOption.value];
+      this.selectedAndEnabledOptions = [this.lastSelectedAndEnabledOption];
```

Comment:
> Tags now appear in the order that Dropdown Options are selected by the user

The solution to the above issue was this field: an array of selected options that's kept in sync with  and ordered the same as `this.value`. Before, I was iterating through [`this.selectedOptions`](https://github.com/CrowdStrike/glide-core/pull/932/files#diff-7431ab16692f7d7a7cfecdb04f55d6c62f87417591065650d88dc407a9b62f3bL400-L407), which was just a getter that returned selected options in their DOM order.

`this.selectedAndEnabledOptions` is appended to when `this.value` is changed. The drawback, of course, is that there's now one more piece of state that needs to be kept up to date. My hope is that the additional tests and assertions I added above will ensure that it is kept up to date.